numpy/core/src/multiarray/descriptor.c

```diff
@@ -1651,6 +1654,9 @@ PyArray_DescrConverter(PyObject *obj, PyArray_Descr **at)
 
     if (PyDataType_ISUNSIZED(*at) && (*at)->elsize != elsize) {
         PyArray_DESCR_REPLACE(*at);
+        if (*at == NULL) {
+            goto fail;
```

Comment:
This should be `goto error`, not `goto fail`.

As is, it silences the `MemoryError` that might occur here, and tries to replace it with another exception

Reply:
Or perhaps even just `return NPY_FAIL`

Reply:
fixed